File: app/src/main/java/ru/cj264/geekbrains/android_intro/homework/ui/notes_list/NotesListViewModelFactory.java
package ru.cj264.geekbrains.android_intro.homework.ui.notes_list;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import ru.cj264.geekbrains.android_intro.homework.domain.FirestoreNotesRepository;
import ru.cj264.geekbrains.android_intro.homework.domain.MockNotesRepository;

public class NotesListViewModelFactory implements ViewModelProvider.Factory {
    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new NotesListViewModel(FirestoreNotesRepository.INSTANCE);
    }
}

File: app/src/main/java/ru/cj264/geekbrains/android_intro/homework/ui/notes_list/NotesListViewModelFactory.kt
package ru.cj264.geekbrains.android_intro.homework.ui.notes_list

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import ru.cj264.geekbrains.android_intro.homework.domain.FirestoreNotesRepository

class NotesListViewModelFactory : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return NotesListViewModel(FirestoreNotesRepository.INSTANCE) as T
    }
}